docs/building-modules/handler.md

```diff
@@ -69,20 +63,19 @@ ctx.EventManager().EmitEvent(
 
 These `events` are relayed back to the underlying consensus engine and can be used by service providers to implement services around the application. Click [here](../core/events.md) to learn more about `events`. 
 
-Finally, the `handler` function returns a `*sdk.Result` which contains the aforementioned `events` and an optional `Data` field. 
+Finally, the invoked `msgServer` method returns a `proto.Message` response and an `error` which are then wrapped in `*sdk.Result` which contains the aforementioned `events` and an optional `Data` field, using `sdk.WrapServiceResult` function. 
```

Comment:
```suggestion
The invoked `msgServer` method returns a `proto.Message` response and an `error` in a `*sdk.Result` object. Finally, the handler creates a `*sdk.Result` and attaches`events` and an optional `Data` field, using `sdk.WrapServiceResult` function using the previously wrapped context. 
```

Reply:
@robert-zaremba 
```The invoked `msgServer` method returns a `proto.Message` response and an `error` _in a `*sdk.Result` object_." ```
This sounds wrong to me, a `msgServer` method in itself returns a proto message response and an `error`, see https://github.com/cosmos/cosmos-sdk/blob/v0.40.0-rc1/x/slashing/handler.go#L19, adding "in a `*sdk.Result` object" sounds confusing to me.

Same for ```using `sdk.WrapServiceResult` function using the previously wrapped context."```
`WrapServiceResult` is using `ctx` from https://github.com/cosmos/cosmos-sdk/blob/v0.40.0-rc1/x/slashing/handler.go#L13, and that results in some never-ending sentence with 2 "using".

Maybe a better option would be:
 The invoked `msgServer` RPC service method returns a `proto.Message` response and an `error`. These return values are then wrapped into an `*sdk.Result` or an `error` using `sdk.WrapServiceResult(ctx sdk.Context, res proto.Message, err error)`. This method takes care of marshaling the `res` parameter to protobuf and attaching any events on the `ctx.EventManager()` to the `sdk.Result`.

Reply:
OK. Better then the the OP and the proposal.